newIDE/app/src/ResourcesList/LocalPiskelBridge.js

```diff
@@ -53,9 +53,7 @@ export const openPiskel = ({
   ipcRenderer.on(
     'piskel-changes-saved',
     (event, outputResources, newAnimationName, receivedData) => {
```

Comment:
@blurymind To give you a clearer idea of what I was thinking about, "receivedData" here is what I consider to be the external editor data, which is then transformed in metadata IF there are `receivedData.data`.

Ideally, we could follow the flow of the metadata like this:

metadata (in GD) ---> external editor data (sent to Piskel) ---> Piskel does whatever you want with this ----> external editor data (sent back to GD) -----> metadata (back in GD).

With the renaming we'll be there I think ;)

Reply:
I think that renaming the variable where the data is stored is making it a little bit more confusing, especially when we also need to consider flow as well.

SO to summarize you want gdevelop to see this as metadata and external editors to see it as 'externalEditorData' ?

Reply:
I mean, if it's storing the same thing, if the "shape" of the object is the SAME, then it should the same.

But here, we have: 
* `receivedData` coming from the editor.
* Then `const externalEditorData = receivedData.data ? { pskl: receivedData } : null;` which is not the same thing because `externalEditorData` is *containing* receivedData. So there is no way these objects are representing the same thing, sot hey should be named differently.

  One is some "data" received from the editor, while the other is an object containing a key named "pskl" and containing the "data" received from the editor.

* Finally we have:
`setMetadata(JSON.stringify(externalEditorData));`

which make me think that the variable named "externalEditorData" here should be renamed to "metadata". receivedData can stay the same.

My rule of thumb is:
* If something is coming from getMetadata or sent to setMetadata, call it "metadata".
* As soon as you store it in another object, or store it with some other things, don't name this other object "metadata". Name if with whatever conveys the best the meaning of what this object is used for.

If the external editor was an editor returning information about how to build a car, we would do:

```
  ipcRenderer.on(
    'car-changes-saved',
    (event, outputResources, newAnimationName, carInformationData) => {
   const metadata = carInformationData.carContent ? { car: carInformationData.carContent } : null;

   if (carInformationData.singleCar) {
     if (metadata) theOnlyResource.setMetadata(metadata);
     onChangesSaved(outputResources, newAnimationName);
   } else {
     onChangesSaved(outputResources, newAnimationName, metadata);
   }

}
```

Reply:
ok I renamed all variables again. Will test and hopefuly no more regressions :)

Reply:
Thanks! I have nothing to comment on the code, will get it a try once I'm done with what I'm working on :)